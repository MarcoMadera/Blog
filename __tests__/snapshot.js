import React from "react";
import renderer from "react-test-renderer";
import Index from "../pages/index";

jest.mock("next/router", () => ({
  useRouter: jest.fn().mockImplementation(() => ({
    query: {
      page: "1",
    },
    beforePopState: require("next/router"),
  })),
}));

it("renders homepage unchanged", () => {
  const tree = renderer.create(<Index />).toJSON();
  expect(tree).toMatchSnapshot();
});
