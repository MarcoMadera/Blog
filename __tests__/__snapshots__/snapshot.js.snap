// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<main
  className="jsx-1696364657 jsx-2525118743"
  id="main"
>
  <aside
    className="jsx-2401816559 jsx-3366886839"
  >
    <picture
      className="jsx-2401816559 jsx-3366886839"
    >
      <source
        className="jsx-2401816559 jsx-3366886839"
        media="(max-width: 876px)"
        srcSet="/profile-222x222.jpg"
      />
      <img
        alt="Marco Madera"
        className="jsx-2401816559 jsx-3366886839"
        height="80"
        loading="eager"
        src="/profile-80x80.jpg"
        width="80"
      />
    </picture>
    <p
      className="jsx-2401816559 jsx-3366886839"
    >
      Â¡Hola!
       
      <span
        aria-label="emoji mano saludando"
        className="jsx-2401816559 jsx-3366886839"
        role="img"
      >
        ðŸ‘‹
      </span>
       
      Soy Marco, autor del blog. Gracias por pasarte, cualquier cosa me puedes contactar a travÃ©s de las siguientes redes:
    </p>
    <a
      aria-label="PÃ¡gina de Github"
      className="jsx-2401816559 jsx-3366886839"
      href="https://github.com/MarcoMadera"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        fill="#b50000"
        height={30}
        viewBox="0 -3.5 256 256"
        width={30}
      >
        <title>
          Github
        </title>
        <path
          d="M127.505 0C57.095 0 0 57.085 0 127.505c0 56.336 36.534 104.13 87.196 120.99 6.372 1.18 8.712-2.766 8.712-6.134 0-3.04-.119-13.085-.173-23.739-35.473 7.713-42.958-15.044-42.958-15.044-5.8-14.738-14.157-18.656-14.157-18.656-11.568-7.914.872-7.752.872-7.752 12.804.9 19.546 13.14 19.546 13.14 11.372 19.493 29.828 13.857 37.104 10.6 1.144-8.242 4.449-13.866 8.095-17.05-28.32-3.225-58.092-14.158-58.092-63.014 0-13.92 4.981-25.295 13.138-34.224-1.324-3.212-5.688-16.18 1.235-33.743 0 0 10.707-3.427 35.073 13.07 10.17-2.826 21.078-4.242 31.914-4.29 10.836.048 21.752 1.464 31.942 4.29 24.337-16.497 35.029-13.07 35.029-13.07 6.94 17.563 2.574 30.531 1.25 33.743 8.175 8.929 13.122 20.303 13.122 34.224 0 48.972-29.828 59.756-58.22 62.912 4.573 3.957 8.648 11.717 8.648 23.612 0 17.06-.148 30.791-.148 34.991 0 3.393 2.295 7.369 8.759 6.117 50.634-16.879 87.122-64.656 87.122-120.973C255.009 57.085 197.922 0 127.505 0"
        />
        <path
          d="M47.755 181.634c-.28.633-1.278.823-2.185.389-.925-.416-1.445-1.28-1.145-1.916.275-.652 1.273-.834 2.196-.396.927.415 1.455 1.287 1.134 1.923m6.272 5.596c-.608.564-1.797.302-2.604-.589-.834-.889-.99-2.077-.373-2.65.627-.563 1.78-.3 2.616.59.834.899.996 2.08.36 2.65m4.304 7.159c-.782.543-2.06.034-2.849-1.1-.781-1.133-.781-2.493.017-3.038.792-.545 2.05-.055 2.85 1.07.78 1.153.78 2.513-.019 3.069m7.277 8.292c-.699.77-2.187.564-3.277-.488-1.114-1.028-1.425-2.487-.724-3.258.707-.772 2.204-.555 3.302.488 1.107 1.026 1.445 2.496.7 3.258m9.403 2.8c-.307.998-1.741 1.452-3.185 1.028-1.442-.437-2.386-1.607-2.095-2.616.3-1.005 1.74-1.478 3.195-1.024 1.44.435 2.386 1.596 2.086 2.612m10.703 1.187c.036 1.052-1.189 1.924-2.705 1.943-1.525.033-2.758-.818-2.774-1.852 0-1.062 1.197-1.926 2.721-1.951 1.516-.03 2.758.815 2.758 1.86m10.514-.403c.182 1.026-.872 2.08-2.377 2.36-1.48.27-2.85-.363-3.039-1.38-.184-1.052.89-2.105 2.367-2.378 1.508-.262 2.857.355 3.049 1.398"
        />
      </svg>
    </a>
    <a
      aria-label="PÃ¡gina de LinkedIn"
      className="jsx-2401816559 jsx-3366886839"
      href="https://www.linkedin.com/in/marcomadera"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        fill="#b50000"
        height={30}
        viewBox="0 0 504.4 504.4"
        width={30}
      >
        <title>
          LinkedIn
        </title>
        <path
          d="M377.6.2H126.4C56.8.2 0 57 0 126.6v251.6c0 69.2 56.8 126 126.4 126H378c69.6 0 126.4-56.8 126.4-126.4V126.6C504 57 447.2.2 377.6.2zM168 408.2H96v-208h72v208zm-36.4-240c-20.4 0-36.8-16.4-36.8-36.8s16.4-36.8 36.8-36.8 36.8 16.4 36.8 36.8c-.4 20.4-16.8 36.8-36.8 36.8zm276.8 240H348V307.4c0-24.4-3.2-55.6-36.4-55.6-34 0-39.6 26.4-39.6 54v102.4h-60v-208h56v28h1.6c8.8-16 29.2-28.4 61.2-28.4 66 0 77.6 38 77.6 94.4v114z"
        />
      </svg>
    </a>
    <a
      aria-label="PÃ¡gina de Twitter"
      className="jsx-2401816559 jsx-3366886839"
      href="https://twitter.com/madera_marco"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        fill="#b50000"
        height={30}
        viewBox="0 0 410.155 410.155"
        width={30}
      >
        <title>
          Twitter
        </title>
        <path
          d="M403.632 74.18a162.414 162.414 0 01-28.28 9.537 88.177 88.177 0 0023.275-37.067c1.295-4.051-3.105-7.554-6.763-5.385a163.188 163.188 0 01-43.235 17.862 11.02 11.02 0 01-2.702.336c-2.766 0-5.455-1.027-7.57-2.891-16.156-14.239-36.935-22.081-58.508-22.081-9.335 0-18.76 1.455-28.014 4.325-28.672 8.893-50.795 32.544-57.736 61.724-2.604 10.945-3.309 21.9-2.097 32.56a3.166 3.166 0 01-.797 2.481 3.278 3.278 0 01-2.753 1.091c-62.762-5.831-119.358-36.068-159.363-85.14-2.04-2.503-5.952-2.196-7.578.593-7.834 13.44-11.974 28.812-11.974 44.454 0 23.972 9.631 46.563 26.36 63.032a79.24 79.24 0 01-20.169-7.808c-3.06-1.7-6.825.485-6.868 3.985-.438 35.612 20.412 67.3 51.646 81.569a79.567 79.567 0 01-16.786-1.399c-3.446-.658-6.341 2.611-5.271 5.952 10.138 31.651 37.39 54.981 70.002 60.278-27.066 18.169-58.585 27.753-91.39 27.753l-10.227-.006c-3.151 0-5.816 2.054-6.619 5.106-.791 3.006.666 6.177 3.353 7.74 36.966 21.513 79.131 32.883 121.955 32.883 37.485 0 72.549-7.439 104.219-22.109 29.033-13.449 54.689-32.674 76.255-57.141 20.09-22.792 35.8-49.103 46.692-78.201 10.383-27.737 15.871-57.333 15.871-85.589v-1.346c-.001-4.537 2.051-8.806 5.631-11.712a174.776 174.776 0 0035.16-38.591c2.573-3.849-1.485-8.673-5.719-6.795z"
        />
      </svg>
    </a>
  </aside>
  <section
    className="jsx-1696364657 jsx-2525118743"
  >
    <h1
      className="jsx-1696364657 jsx-2525118743"
    >
      Ãšltimos artÃ­culos
    </h1>
    <main
      className="jsx-4210035564"
      id="main"
    >
      <h1
        className="jsx-4210035564"
      >
        404
      </h1>
      <p
        className="jsx-4210035564"
      >
        PÃ¡gina no encontrada
      </p>
    </main>
    <nav
      aria-label="PaginaciÃ³n"
      className="jsx-1696364657 jsx-2525118743"
    >
      <ol
        className="jsx-1696364657 jsx-2525118743"
      />
    </nav>
  </section>
  <aside
    className="jsx-1696364657 jsx-2525118743"
  >
    <section
      className="jsx-1238778292"
    >
      <h2
        className="jsx-1238778292"
      >
        Todas las etiquetas
      </h2>
      <div
        className="jsx-1238778292"
      />
    </section>
    <form
      action="https://buttondown.email/api/emails/embed-subscribe/MarcoMadera"
      className="jsx-670821710"
      method="post"
      noValidate={true}
      onSubmit={[Function]}
      target="_blank"
    >
      <label
        className="jsx-670821710"
        htmlFor="bd-email"
      >
        Â¡SuscrÃ­bete al Newsletter!
      </label>
      <p
        className="jsx-670821710"
      >
        RecibirÃ¡s actualizaciones del blog con temas de programaciÃ³n
      </p>
      <input
        className="jsx-670821710"
        id="bd-email"
        name="email"
        onChange={[Function]}
        placeholder="Correo electrÃ³nico*"
        style={
          Object {
            "outline": "unset",
          }
        }
        type="email"
      />
      <input
        className="jsx-670821710"
        name="embed"
        type="hidden"
        value="1"
      />
      <button
        className="jsx-670821710 btn btn-primary"
      >
        SuscrÃ­bete
      </button>
    </form>
  </aside>
</main>
`;
